autopush/db.py

```diff
@@ -145,22 +139,25 @@ def make_rotating_tablename(prefix, delta=0, date=None):
     return "{}_{:04d}_{:02d}".format(prefix, date.year, date.month)
 
 
-def create_rotating_message_table(prefix="message", delta=0, date=None,
-                                  read_throughput=5,
-                                  write_throughput=5):
-    # type: (str, int, Optional[datetime.date], int, int) -> Table
+def create_rotating_message_table(
+        prefix="message",     # type: str
+        delta=0,              # type: int
+        date=None,            # type: Optional[datetime.date]]
+        read_throughput=5,    # type: int
+        write_throughput=5,   # type: int
+        boto_resource=None):  # type: DynamoDBResource
+    # type: (...) -> DynamoDBTable
     """Create a new message table for webpush style message storage"""
     tablename = make_rotating_tablename(prefix, delta, date)
 
     try:
-        table = g_dynamodb.Table(tablename)
-        if table.table_status == 'ACTIVE':  # pragma nocover
+        table = DynamoDBTable(boto_resource, tablename)
+        if table.table_status == 'ACTIVE':
             return table
     except ClientError as ex:
         if ex.response['Error']['Code'] != 'ResourceNotFoundException':
-            # If we hit this, our boto3 is misconfigured and we need to bail.
-            raise ex  # pragma nocover
```

Comment:
Might we still want to raise exceptions here (I'm not sure)? Either way this block of code isn't necessary any longer as it's a "pass" for any exception

Reply:
ah! Good catch, thanks!